velox/connectors/hive/TableFormatReader.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/connectors/hive/HiveConnectorSplit.h"
+#include "velox/connectors/hive/TableHandle.h"
+#include "velox/dwio/common/BufferedInput.h"
+#include "velox/dwio/common/Reader.h"
+
+namespace facebook::velox::connector::hive {
+
+static const char* kPath = "$path";
+static const char* kBucket = "$bucket";
+
+class TableFormatReader {
```

Comment:
Should the class be `TableReader` instead of `TableFormatReader`?

Reply:
`SplitReader` makes sense too since that is the unit of processing here. `TableFormatReader` is too verbose.

Reply:
I was thinking to rename FormatData to FileFormatReader, so changed the TableReader to TableFormatReader to make the names aligned. Do you have strong preference on TableReader over TableFormatReader?

Reply:
Ok, renamed to SplitReader